--- a/third_party/WebKit/Source/wtf/LinkedHashSet.h	2017-11-24 02:03:48.045943722 -0500
+++ b/third_party/WebKit/Source/wtf/LinkedHashSet.h	2017-11-24 02:04:34.212508077 -0500
@@ -687,6 +687,8 @@
   return *this;
 }

+inline void swapAnchor(LinkedHashSetNodeBase& a, LinkedHashSetNodeBase& b);
+
 template <typename T, typename U, typename V, typename W>
 inline void LinkedHashSet<T, U, V, W>::swap(LinkedHashSet& other) {
   m_impl.swap(other.m_impl);
--- a/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h.bak	2017-11-24 13:42:38.842944797 -0500
+++ b/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h	2017-11-24 13:43:06.219524249 -0500
@@ -4,7 +4,7 @@

 #include "platform/PlatformExport.h"
 #include "wtf/ThreadSpecific.h"
-
+#include <functional>
 #include <memory>

 namespace gpu {
